@here/harp-mapview/lib/Utils.ts

```diff
@@ -223,40 +225,62 @@ export namespace MapViewUtils {
      */
-    export function orbitFocusPoint(
+    export function orbitAroundScreenPoint(
         mapView: MapView,
-        deltaAzimuthDeg: number,
-        deltaTiltDeg: number,
-        maxTiltAngleRad = MAX_TILT_RAD
+        offsetX: number,
+        offsetY: number,
+        deltaAzimuth: number,
+        deltaTilt: number,
+        maxTiltAngle: number
     ) {
-        const target = mapView.target;
-        const sphericalCoordinates = extractSphericalCoordinatesFromLocation(
+        const rotationTargetWorld = MapViewUtils.rayCastWorldCoordinates(mapView, offsetX, offsetY);
+        if (rotationTargetWorld === null) {
+            return;
+        }
+
+        const headingAxis = mapView.projection.surfaceNormal(rotationTargetWorld, cache.vector3[1]);
+        const headingQuat = cache.quaternions[1].setFromAxisAngle(headingAxis, -deltaAzimuth);
+        mapView.camera.quaternion.premultiply(headingQuat); // applyMatrix4(headingTransform);
+        mapView.camera.position.sub(rotationTargetWorld);
+        mapView.camera.position.applyQuaternion(headingQuat);
+        mapView.camera.position.add(rotationTargetWorld);
+
+        const mapTargetWorld =
+            offsetX === 0 && offsetY === 0
+                ? rotationTargetWorld
+                : MapViewUtils.rayCastWorldCoordinates(mapView, 0, 0);
+        if (mapTargetWorld === null) {
+            return;
+        }
+
+        const { tilt: currentTilt } = MapViewUtils.extractSphericalCoordinatesFromLocation(
             mapView,
             mapView.camera,
-            target
-        );
-        const tilt = Math.max(
-            Math.min(
-                THREE.MathUtils.radToDeg(maxTiltAngleRad),
-                deltaTiltDeg + THREE.MathUtils.radToDeg(sphericalCoordinates.tilt)
-            ),
-            0
+            mapTargetWorld
         );
-        const heading =
-            THREE.MathUtils.radToDeg(sphericalCoordinates.azimuth + Math.PI) + deltaAzimuthDeg;
-        mapView.lookAt({
-            target,
-            distance: mapView.targetDistance,
-            tilt,
-            heading
-        });
+
+        //FIXME: For globe tilt in the map center is different from the tilt in the rotation center,
+        //hence the clamped tilt is too conservative.
+        const clampedDeltaTilt =
+            MathUtils.clamp(deltaTilt + currentTilt, 0, maxTiltAngle) - currentTilt;
+        if (Math.abs(clampedDeltaTilt) <= Number.EPSILON) {
+            return;
+        }
+        const tiltAxis = cache.vector3[0].set(1, 0, 0).applyQuaternion(mapView.camera.quaternion);
+        const tiltQuat = cache.quaternions[0].setFromAxisAngle(tiltAxis, clampedDeltaTilt);
+        mapView.camera.quaternion.premultiply(tiltQuat);
+        mapView.camera.position.sub(rotationTargetWorld);
+        mapView.camera.position.applyQuaternion(tiltQuat);
+        mapView.camera.position.add(rotationTargetWorld);
```

Comment:
This can put the target out of the world. After that the orbit controls won't work anymore since we early exit above. It can be reproduced in your build both in sphere and mercator by increasing the maximum tilt angle and then tilting to max around and arbitrary screen point.

Reply:
The problem is, that the `MapControls.maxTiltAngle` is handled inconsistently.
Previously the maximum tilt was clamped to 90 degrees but internally it was never possible to go beyond 89 degrees. I will check why the clamping is lost and will clamp consistently to 89 degrees.

Reply:
found the culprit: `this.m_maxTiltAngle = Math.max(0, Math.min(90, THREE.MathUtils.degToRad(angle)));` So the angle was clamped to 90 degrees after converting to radians.
It works now, but still for globe we can get into a situation where the target is not existing anymore. I will investigate why this is happening.